charts/newrelic-infrastructure-v3/templates/NOTES.txt

```diff
@@ -4,13 +4,112 @@ Your deployment of the New Relic Infrastructure agent is complete if you have no
 
 {{- if or (not .Values.ksm.enabled) (not .Values.kubelet.enabled) }}
 
-###########
-# WARNING #
-###########
-
+-> WARNING LEGACY CONFIG<-
 You have specified ksm or kubelet integration components as not enabled.
 Those components are needed to have the full experience on NROne kubernetes explorer.
+{{- end }}
+
+{{- if and .Values.nodeAffinity .Values.controlPlane.enabled (not (include "newrelic.fargate" .)) }}
+
+-> WARNING LEGACY CONFIG <-
+You have specified the legacy "nodeAffinity" option into values, however the new version of the chart
+supports a different nodeAffinity options per component.
+
+The value specified has been applied automatically to ksm and kubelet workloads, however **NOT** to the control plane one.
+
+Keep in mind you should leverage the new ksm.affinity.nodeAffinity, controlPlane.affinity.nodeAffinity
+and kubelet.affinity.nodeAffinity to customize nodeAffinity.
+{{- end }}
+
+{{- if .Values.resources }}
+
+-> WARNING LEGACY CONFIG <-
+You have specified the legacy "resources" option into values, however the new version of the chart
+supports different resources per component, therefore the value specified has been ignored.
+
+Please use ksm.resources, controlPlane.resources and kubelet.resources.
+{{- end }}
+
+{{- if .Values.tolerations }}
+
+-> WARNING LEGACY CONFIG <-
+You have specified the legacy "tolerations" options into values, however the new version of the chart
+supports different tolerations per component, therefore the value specified has been ignored.
+
+Please use ksm.tolerations, controlPlane.tolerations and kubelet.tolerations.
+{{- end }}
+
+{{- if .Values.image }}
+
+-> WARNING LEGACY CONFIG <-
+You have specified into values one of the legacy options "image.*".
+The following values are no longer supported and are currently ignored.
+ - image.repository
+ - image.tag
+ - image.pullPolicy
+ - image.pullSecrets
+
+Notice that the 3.x version of the integration uses 3 different images.
+Please set:
+ - images.forwarder.* to configure the image in charge of sending data to newrelic backend
+ - images.agent.* to configure the image bundling the agent and onHost integration
+ - images.integration.* to configure the image in charge of scraping k8s data
+{{- end }}
+
+{{- if (or .Values.enableWindows .Values.windowsOsList .Values.windowsSecurityContext .Values.windowsNodeSelector)}}
+
+-> WARNING LEGACY CONFIG <-
+Windows is currently supported by 2.x charts only and therefore "windowsOsList" "windowsSecurityContext" "windowsNodeSelector"
+are no longer supported.
+
+{{- end }}
```

Comment:
If we cannot shim the values in an acceptable way I think we should fail so customer does not end up with unexpected config:

```suggestion
{{- if (or .Values.enableWindows .Values.windowsOsList .Values.windowsSecurityContext .Values.windowsNodeSelector)}}

-> WARNING LEGACY CONFIG <-
Windows is currently supported by 2.x charts only and therefore "windowsOsList" "windowsSecurityContext" "windowsNodeSelector"
are no longer supported.

{{- fail "Windows nodes are not supported in nri-kubernetes v3" }}
{{- end }}
```

Reply:
That will not work. That you are suggesting will show `Windows nodes are not supported in nri-kubernetes v3` without the message above.

You have to
```
{{- fail "
Windows is currently supported by 2.x charts only and therefore "windowsOsList" "windowsSecurityContext" "windowsNodeSelector"
are no longer supported.
" }}

Reply:
If we are going to fail, then the nit I've put about having a helper which renders all the errors at the same time is more useful to don't hammer the user with errors showing them one by one while he is solving them.

Reply:
Fuck, you're totally right. For some reason I though it would have been rendered anyway :(

Reply:
We could do some helm magic:
 - Save a variable `errors` and append there one by one all the errors and merely at the end print them at once with the fail
 
 It is a bit hackish and honestly I prefer the long message explaining what is going on, what do you think?

Reply:
This works:
```
{{ $errors:= "" }}

{{- if .Values.logFile }}
{{ $errors = printf "%s%s" $errors (include "newrelic.compatibility.message.log" . | nindent 2) }}
{{- end }}

{{- if .Values.apiServerSecurePort }}
{{- if .Values.logFile }}
{{ $errors = printf "%s%s" $errors (include "newrelic.compatibility.message.apiServerSecurePort" . | nindent 2) }}
{{- end }}
{{- end }}

{{- if $errors }}
{{- fail $errors }}
{{- end }}
```

Makes sense to implement it?

Reply:
I gave a shot, let me know if it makes sense. I am not failing for every condition and for some just printing a warning

Reply:
Looking pretty nice to me!